src/forklift/arcgis.py

```diff
@@ -61,23 +62,27 @@ def ensure(self, what, affected_services):
                      'No services will be stopped or started. See README.md for more details.')
             return (True, None)
 
-        while len(affected_services) > 0 and tries >= 0:
-            problem_child = []
-            sleep(wait[tries])
-
-            for service_name, service_type in affected_services:
-                if what == 'off':
-                    log.debug('stopping %s.%s', service_name, service_type)
-                    status, message = self.turn_off(service_name, service_type)
-                else:
-                    log.debug('starting %s.%s', service_name, service_type)
-                    status, message = self.turn_on(service_name, service_type)
+        def act_on_service(service_info):
+            service_name, service_type = service_info
+            if what == 'off':
+                log.debug('stopping %s.%s', service_name, service_type)
```

Comment:
Good catch. It doesn't show up at all at the command line or in the log. I'll log `affected_services` outside of this function.